src/main/java/com/example/board/dto/PostResponseDto.java

```diff
@@ -0,0 +1,31 @@
+package com.example.board.dto;
+
+import com.example.board.exception.BaseException;
+import com.example.board.exception.ErrorMessage;
+import com.example.board.model.Post;
+
+public record PostResponseDto(
+        Long postId,
+        String userName,
+        String title
+) {
+    public static PostResponseDto from(Post post) {
+        postResponseValidate(post);
+        return new PostResponseDto(post.getId(), post.getTitle(), post.getUser().getName());
```

Comment:
실수로 순서를 잘못 넣으신 것 같습니다! 저도 이럴 때가 종종 있는데 이런 의미에서 빌더 패턴은 좋은 것 같습니다!
```suggestion
        return new PostResponseDto(post.getId(), post.getUser().getName(), post.getTitle());
```

Reply:
헉 정말 감사합니다ㅠㅠ 빌더 패턴을 한번도 써본 적이 없었는데 더 찾아봐야겠군요!

Reply:
builder 패턴에 대해 의문이 있어 얘기해보고 싶어요! 개인적으로 빌더패턴은 서비스 로직에 도메인 생성자 로직이 너무 노출이 많 느낌이 있습니다. 실수를 줄여주는 느낌이 있기는 한데 서비스에 너무 생성자 로직이 보이는 게 그렇게 좋은 건가 하는 느낌을 지우기 힘들었습니다.
의견을 들어보고 싶어요:)

Reply:
빌더가 옳은 방법이라기 보단 실수를 줄여주는데 도움이 된다고 생각해서 이럴 때 생각해보면 빌더 패턴도 좋을 것 같다고 말씀드리고 싶었습니다! 저도 빌더를 안 써봤다가 이번 과제 때 써보면서 이럴 때는 좋겠구나 하고 체감이 있었습니다.

저는 개인적으로 생성의 의도를 네이밍에 담을 여지가 있는 정적 팩토리 메서드를 조금 선호하던 편인 것 같습니다. 말씀하신대로 생성자나 혹은 정적 팩토리 메서드를 사용해서 캡슐화를 더욱 지킬 수 있다고 생각합니다. 예를 들면 지금 같은 PostResponseDto도 생성자 파라미터에 Post를 받아 생성할 수 있으니, 서비스 레이어에선 그저 넘겨주기만 하면 되겠죠!

파라미터가 늘어나고 생성 방식이 늘어나면 도메인이 커지게 되는건 항상 마음에 걸렸던 것 같습니다. 많은 일을 하는 것 같진 않은데 생성자만으로 몇십 줄을 채우니 도메인 클래스가 너무 뚱뚱해져서 정적 팩토리 메서드를 선호하면서도 마음에 걸렸었는데, 결국 생성에 대한 책임을 별개의 책임으로 보고 분리할까? 하고 고민된 적도 있었는데요...

이번에 빌더 패턴을 사용하고 고려해보면서 빌더를 통해 생성의 책임을 분리할 수 있다고 생각이 들었습니다. 즉, 빌더 패턴을 사용하면 도메인 클래스를 작게 유지 할 수 있고, 파라미터 실수 방지는 덤이라고 생각했습니다.

물론 도메인이 크지 않을 땐 과하다고 생각합니다! 파라미터가 늘어날 수록 책임을 분리하고 유지보수 하는데 도움이 될 수도 있겠다고 생각이 들지만, 생성의 책임을 분리하지 않는 것이 낫다고 여기시거나 OOP의 절대적인 특성인 캡슐화가 더 중요하다 여기시면 빌더 패턴을 아예 고려하지 않는 것이 바람직하다고 생각합니다. 저도 앞으로 다양하게 사용해보면서 애플리케이션을 키웠을 때 선호하는 방향이 명확해질 것 같습니다. 책임은 추상적인 개념이고 객체지향은 확장과 유지보수를 위한 것이기 때문이니까요🤩